client/file_logging.py

```diff
@@ -0,0 +1,138 @@
+	description = """
+	Keep logs of campaign feedback and results. The directory
+	of the logged file(s), the file size (in MB) of the log, and
+	the number of log files kept can be specified. Plugin requires
+	the 'logging' and 'os' packages available for Python.
+	"""
+	req_packages = {
+		'logger': has_logger,
+		'os': has_os
+	}
+	homepage = 'https://github.com/securestate/king-phisher-plugins'
+	options = [
+		plugins.ClientOptionString(
+			'log_dir',
+			'The directory in which to create the log file(s).',
+			default=DEFAULT_LOG_DIR,
+			display_name='Log Directory'
+		),
+		plugins.ClientOptionInteger(
+			'log_file_size',
+			'The maximum size, in megabytes, of a single log file.',
+			default=DEFAULT_LOG_FILE_SIZE,
+			display_name='Log File Size (MB)'
+		),
+		plugins.ClientOptionInteger(
+			'log_file_count',
+			'The number of log files that will be maintained by the plugin.',
+			default=DEFAULT_LOG_FILE_COUNT,
+			display_name='Number of Log Files'
+		)
+	]
+
+	# this is the primary plugin entry point which is executed when the plugin is enabled
+	def initialize(self):
+		log_dir = self.config['log_dir']
+		unsanitary_log_dir_arg = False
+
+		# attempt to create the directory for the log file if it does not exist
+		try:
+			if not os.path.exists(log_dir):
+				os.mkdir(log_dir)
+		except Exception:
+			# revert to default directory
+			unsanitary_log_dir_arg = True
+			self.config['log_dir'] = DEFAULT_LOG_DIR
+			log_dir = DEFAULT_LOG_DIR
+
+			# create the directory if it does not exist
+			if not os.path.exists(log_dir):
+				os.mkdir(log_dir)
+
+		# sanitize the log file size option argument
+		log_file_size = self.config['log_file_size']
+		if log_file_size < MIN_LOG_FILE_SIZE:
+			unsanitary_file_size_arg = True
+			self.config['log_file_size'] = MIN_LOG_FILE_SIZE
+			log_file_size = MIN_LOG_FILE_SIZE
+		else:
+			unsanitary_file_size_arg = False
+
+		# sanitize the log file count option argument
+		log_file_count = self.config['log_file_count']
+		if log_file_count < MIN_LOG_FILE_COUNT:
+			unsanitary_file_count_arg = True
+			self.config['log_file_count'] = MIN_LOG_FILE_COUNT
+			log_file_size = MIN_LOG_FILE_COUNT
+		else:
+			unsanitary_file_count_arg = False
+
+		# convert the specified log file size (MB) to bytes for use by the logger
+		file_size = log_file_size * 1024 * 1024
+
+		# grab the logger in use by the client (root logger)
+		logger = logging.getLogger(LOGGER_NAME)
+
+		# set up the handler and formatter for the logger, and attach the components
+		handler = logging.handlers.RotatingFileHandler("{0}/{1}".format(log_dir, LOG_FILE_NAME), maxBytes=file_size, backupCount=log_file_count)
+		formatter = logging.Formatter('%(asctime)s -- %(name)s -- %(levelname)s: %(message)s')
+		handler.setFormatter(formatter)
+		logger.addHandler(handler)
+
+		# keep reference of handler as an attribute
+		self.handler = handler
+
+		# Set level of logger to accept up to debug info
+		logger.setLevel(logging.DEBUG)
```

Comment:
There should no reason to set it, as this is handled by client when starting up. You should only need to attach your handler to it.